test/extended/operators/olm.go

```diff
@@ -168,4 +169,35 @@ var _ = g.Describe("[Feature:Platform] an end user use OLM", func() {
 		o.Expect(err).NotTo(o.HaveOccurred())
 		o.Expect(output).To(o.ContainSubstring(etcdClusterName))
 	})
+
+	// OCP-23395 - Deleted catalog registry pods and verify if them are recreated automatically
+	// author: bandrade@redhat.com
+	g.It("Recreate catalog registry pods and if they are deleted", func() {
+		msg, err := oc.SetNamespace("openshift-marketplace").AsAdmin().Run("get").Args("operatorsource", "-o=jsonpath={range .items[*].metadata}{.name}{'\\n'}").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		catalogSources := strings.Split(msg, "\n")
+		e2e.Logf("Catalog sources: %s", catalogSources)
+		if len(catalogSources) > 0 {
+			catalogSourceLabel := "marketplace.operatorSource=" + catalogSources[0]
+			msg, err := oc.SetNamespace("openshift-marketplace").AsAdmin().Run("delete").Args("pods", "-l "+catalogSourceLabel).Output()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			o.Expect(msg).To(o.ContainSubstring("deleted"))
+
+			poolErr := wait.Poll(5*time.Second, 60*time.Second, func() (bool, error) {
+				msg, err := oc.SetNamespace("openshift-marketplace").AsAdmin().Run("get").Args("pods", "-l "+catalogSourceLabel, "-o=jsonpath={range .items[*].status}{.phase}").Output()
+				if err != nil {
+					e2e.Logf("Fail to get catalogsource pod, error: %v", err)
+					return false, err
```

Comment:
This will exit polling if marketplace if _any_ error is thrown -- are we sure that a get that finds nothing doesn't return an error? If it does, then if marketplace doesn't recreate the pod before this executes, we'll immediately fail out.

Reply:
Makes sense, you suggest to `return false, nil` in that case?

Reply:
Nvm, It already has a block for dealing with failure scenarios, I will remove this block